src/UI/Article.jsx

```diff
@@ -1,26 +1,103 @@
-import styled from "styled-components";
+function Article(props) {
+  const getTitle = props => {
+    if (!props?.data) return 'No Title Available';
 
-const NewsCard = styled.div`
-  background-color: lightgrey;
-  padding: 1.5rem;
-  margin-top: 1rem;
+    return (
+      props.data.headline?.main || props.data.title || 'No Title Available'
+    );
+  };
 
-  display: flex;
-  flex-direction: column;
-  text-align: center;
-`;
-function Article(props) {
-  const { title, byline } = props.props;
-  const content = props.props.abstract || props.props.content;
+  const getContent = props => {
+    if (!props?.data) return 'No Content Available';
+
+    return props.data.abstract || props.data.content || 'No Content Available';
+  };
+
+  const getAuthor = props => {
+    if (!props?.data) return 'Unknown author';
+
+    return (
+      props.data.byline?.original ||
+      props.data.author ||
+      props.data.byline ||
+      'Unknown author'
+    );
+  };
+
+  const getImage = props => {
+    if (!props?.data) return null;
+
+    return (
+      props.data.multimedia?.[0]?.url ||
+      props.data.urlToImage ||
+      props.data.image ||
+      null
+    );
+  };
+
+  const getDate = props => {
+    if (!props?.data) return 'Unknown Date';
+
+    return (
+      props.data.published_date ||
+      props.data.publishedAt ||
+      props.data.pub_date ||
+      'Unknown Date'
+    );
+  };
+
+  const getSource = props => {
+    if (!props?.data) return null;
+
+    return props.data.source?.name || props.data.source || null;
+  };
+
+  const formatDate = dateString => {
+    if (!dateString || dateString === 'Unknown Date') return 'Unknown Date';
+    const date = new Date(dateString);
+    return date.toLocaleDateString('en-GB');
+  };
+
+  // Get values
+  const title = getTitle(props);
+  const content = getContent(props);
+  const author = getAuthor(props);
+  const image = getImage(props);
+  const date = formatDate(getDate(props));
+  const source = getSource(props);
 
   return (
-    <NewsCard>
-      <h1>{title}</h1>
-      <p>{content}</p>
-      <span>
-        <p>{byline}</p>
-      </span>
-    </NewsCard>
+    <div
+      className={`relative flex min-h-[300px] items-end justify-center overflow-hidden rounded bg-slate-600 shadow-md ${image ? '' : 'bg-gradient-to-r from-sky-500 to-indigo-500'}`}
+      style={{
+        backgroundImage: image ? `url(${image})` : '',
+        backgroundSize: 'cover',
+        backgroundPosition: 'center',
+      }}
+    >
+      <div className="absolute bottom-0 left-0 right-0 h-1/2 bg-black/30 p-4 pb-0 backdrop-blur-lg">
+        <h1 className="line-clamp-1 font-bold text-yellow-500">{title}</h1>
+
+        <hr className="my-2w h-1 border-0 bg-yellow-500 opacity-100" />
+
+        <div className="relative">
+          <p className="text-[clamp(0.875rem, 1vw + 0.5rem, 1.25rem)] line-clamp-2 text-white">
```

Comment:
I think, whenever doing clamp, or box shadow or any style that has multiple values (and so the spaces `' '`, tailwind will not pick it up, in these cases you need to replace spaces with underscores, like this:

`text-[clamp(0.875rem, 1vw + 0.5rem, 1.25rem)]`

to this (also not sure the `+` will work like this, think you need `calc()` here)

`text-[clamp(0.875rem,_calc(1vw_+_0.5rem),_1.25rem)]`